cli/azd/resources/starter/azure.yaml

```diff
@@ -0,0 +1,29 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+
+# This is an example starting azure.yaml file containing several services.
+# Make changes as needed to describe your application setup.
+
+# Name of the application.
+name: azd-starter
```

Comment:
@savannahostrowski Let me know if this looks good to you.

I wanted to add `funcapp` and `containerapp` examples, but they seem too complicated to be described simply. Ideally, we'd have documentation that walks through the example and is linked here. Let me know if that is needed.

Reply:
I like the comments here. Wonder if it's worth adding a link to https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/azd-schema at the top so that folks can get familiarized with our schema specifics?

I got a lot of feedback at PyCon that folks needed more template guidance.

Reply:
That is great reference! I think I'll also add commented out section for the template metadata

Reply:
So interesting observation I have seen in multiple azdified azd templates - folks have been including an azure.yaml file but it's empty. Because they are just deploying the app as pre-build Docker images to ACA.

Reply:
Can you link an example? I don't think `azd` would function without a valid `azure.yaml` config. Which "valid" here should mean has `name:<projName>` and maybe empty services.  I certainly agree that users should be able to have an infrastructure-only repository that only works with `azd provision`.

Reply:
I was skeptical too but azd works with an empty azure.yaml. Here's a sample: https://github.com/microsoft/azd-aca-templates.

Reply:
Ah. They skip `deploy`. I think `deploy` ends up failing due to specific checks we've added to ensure non-zero services deployed (although that may have been removed recently).